public/app/features/alerting/unified/hooks/useProduceNewAlertmanagerConfig.ts

```diff
@@ -0,0 +1,64 @@
+import { Action } from '@reduxjs/toolkit';
+import reduceReducers from 'reduce-reducers';
```

Comment:
This might raise some eyebrows so I'll elaborate here for posterity;

Redux's `combineReducers()` will create a _nested_ data structure isolating each reducer by a given "key". This is not what we want, we want separate reducers to modify the _same_ data structure – this is what `reduce-reducers` allows us to achieve.

See [this SO answer](https://stackoverflow.com/questions/38652789/correct-usage-of-reduce-reducers/44371190#44371190).